pkg/endpoint/metrics.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/pkg/metrics"
+)
+
+const (
+	labelEndpointRegenerationBPFCompilation = "bpf-compilation"
+	labelEndpointProxyConfiguration         = "proxy-configuration"
+	labelEndpointRegenerationTotal          = "total"
+)
+
+// regenerationMetric is a helper function to update metrics.EndpointRegenerationTime
+func regenerationMetric(scope string, duration time.Duration) {
+	metrics.EndpointRegenerationTime.With(map[string]string{
+		metrics.LabelScope: scope,
+	}).Observe(duration.Seconds())
+	return
+}
+
+// metricsEndpointRegenerationBPF is a helper function to update
+// metrics.EndpointRegenerationTime with label BPF.
+func (e *Endpoint) metricsEndpointRegenerationBPF(duration time.Duration) {
```

Comment:
Since these functions don't rely on endpoint, it might be nice to make them normal functions. Their dependencies would be a little clearer. It might also be useful to move them to the metrics package, but I'm less sure about that. If we did, we would be saying that the only way to update `EndpointRegenerationTime` is with these functions (we might need to make that metric private then). This might be a good thing.

In any case, I was a little unclear on what the function did until I read it. A verb in the function name might be more clear. `recordEndpointRegeneration*` but then we lose the metric bit.